### PR TITLE
Added changes into core and orienteer-users

### PR DESCRIPTION
**Pull request by WeaxMe**
1. Added changes to OTask micro framework
2. Added hook which modifies _allowRead field into orienteer-users
3. Added Bootstrap footer into AbstractWidget

### CODE REVIEW

File: orienteer-core/src/main/java/org/orienteer/core/tasks/ITaskSession.java [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package org.orienteer.core.tasks;
 
 /**
- * Interface for task sessions 
+ * Interface for task sessions
+ * @param <T> task session type
  */
-public interface ITaskSession {
+public interface ITaskSession<T extends OTaskSession> {
```
Comment (PhantomYdn) [inline orienteer-core/src/main/java/org/orienteer/core/tasks/ITaskSession.java]: That's actually redundunt. You can always to specify more nearow result type in subclasses.

File: orienteer-core/src/main/java/org/orienteer/core/util/CommonUtils.java [hunk 2]
```diff
@@ -85,6 +85,10 @@ public static final CharSequence escapeStringForJSON(CharSequence content) {
 				.collect(Collectors.toList());
 	}
 
+	public static List<ODocument> getDocuments(List<OIdentifiable> identifiables) {
+		return mapIdentifiables(identifiables, d -> d);
```
Comment (PhantomYdn) [inline orienteer-core/src/main/java/org/orienteer/core/util/CommonUtils.java]: OIdentifiable can lead to null. Such records should be excluded.

File: orienteer-users/src/main/java/org/orienteer/users/module/OrienteerUsersModule.java [hunk 3]
```diff
@@ -87,8 +87,8 @@ private void updateOrienteerUserRoleDoc(ODatabaseDocument db, ODocument perspect
 
         role.grant(OSecurityHelper.FEATURE_RESOURCE, SearchPage.SEARCH_FEATURE, OrientPermission.READ.getPermissionFlag());
 
-        role.grant(ResourceGeneric.CLASS, OrienteerUser.CLASS_NAME, OrientPermission.UPDATE.getPermissionFlag());
-        role.grant(ResourceGeneric.DATABASE, "cluster", OrientPermission.UPDATE.getPermissionFlag());
+        role.grant(ResourceGeneric.CLASS, OrienteerUser.CLASS_NAME, 6);
```
Comment (PhantomYdn) [inline orienteer-users/src/main/java/org/orienteer/users/module/OrienteerUsersModule.java]: It's hard to get what 6 actually means. Recommend to use OrientPermission - there is  method to get combined flag as int.

File: orienteer-users/src/main/java/org/orienteer/users/hook/OrienteerUserRoleHook.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import com.orientechnologies.orient.core.db.document.ODatabaseDocument;
+import com.orientechnologies.orient.core.hook.ODocumentHookAbstract;
+import com.orientechnologies.orient.core.metadata.security.ORestrictedOperation;
+import com.orientechnologies.orient.core.metadata.security.ORole;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class OrienteerUserRoleHook extends ODocumentHookAbstract {
```
Comment (PhantomYdn) [inline orienteer-users/src/main/java/org/orienteer/users/hook/OrienteerUserRoleHook.java]: No JavaDoc - it leads to codestyle check fail

File: orienteer-core/src/main/java/org/orienteer/core/util/CommonUtils.java (hunk 2 above)
Comment (WeaxMe) [reply orienteer-core/src/main/java/org/orienteer/core/util/CommonUtils.java]: Fixed in 3a69ccc

File: orienteer-users/src/main/java/org/orienteer/users/hook/OrienteerUserRoleHook.java (hunk 4 above)
Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/hook/OrienteerUserRoleHook.java]: Fixed in 956cc73

File: orienteer-core/src/main/java/org/orienteer/core/tasks/ITaskSession.java (hunk 1 above)
Comment (WeaxMe) [reply orienteer-core/src/main/java/org/orienteer/core/tasks/ITaskSession.java]: Fixed in 37f0106

**Pull request merged**